src/pythermalcomfort/zhang_comfort_model_functions/sensation.py

```diff
@@ -0,0 +1,244 @@
+        Local thermal sensation on sens_third_coldest 9-point/7-point sensation scale [-].
+    """
+
+    def get_coefficient_c1(t_skin_local_array, t_skin_local_set_array):
+        """Return c1_minus or c1_plus based on the comparison of skin temperature to its setpoint."""
+        return np.where(t_skin_local_array - t_skin_local_set_array < 0, c1_minus_array, c1_plus_array)
+
+    def get_coefficient_c2(dt_skin_local_dt_array):
+        """Return c2_minus or c2_plus based on the local skin temperature derivative."""
+        return np.where(dt_skin_local_dt_array < 0, c2_minus_array, c2_plus_array)
+
+    # Set coefficients for the equation
+    c1_plus_array = np.array(list(coefficients.c1_plus.values()))
+    c1_minus_array = np.array(list(coefficients.c1_minus.values()))
+    c1 = get_coefficient_c1(
+        t_skin_local_array=t_skin_local_array,
+        t_skin_local_set_array=t_skin_local_set_array,
+    )
+    k1 = np.array(list(coefficients.k1.values()))
+    c2_plus_array = np.array(list(coefficients.c2_plus.values()))
+    c2_minus_array = np.array(list(coefficients.c2_minus.values()))
+    c2 = get_coefficient_c2(dt_skin_local_dt_array=dt_skin_local_dt_array)
+    c3 = np.array(list(coefficients.c3.values()))
+
+    # Calculate static component of local thermal sensation
+    sensation_local_static_array = 4 * (
+            (
+                    2
+                    / (
+                            1
+                            + np.exp(
+                        - c1 * (t_skin_local_array - t_skin_local_set_array)
+                        - k1
+                        * (
+                                (t_skin_local_array - t_skin_local_set_array)
+                                - (t_skin_mean_value - t_skin_mean_set_value)
+                        )
+                    )
+                    )
+            )
+            - 1
+    )
+
+    # Calculate dynamic component of local thermal sensation
+    sensation_local_dynamic_array = (
+            c2 * dt_skin_local_dt_array + c3 * dt_core_local_dt_array
+    )
+
+    # Calculate local thermal sensation
+    sensation_local_array = (
+            sensation_local_static_array + sensation_local_dynamic_array
+    )
+
+    # Apply the selected thermal sensation scale
+    if sensation_scale == "7-point":
+        # Clip the array between -3 and 3
+        sensation_local_array = np.clip(sensation_local_array, -3, 3)
+    elif sensation_scale == "9-point":
+        # Clip the array between -4 and 4
+        sensation_local_array = np.clip(sensation_local_array, -4, 4)
+    else:
+        raise ValueError("Please select '7-point' or '9-point'")
+
+    return sensation_local_array
+
+
+def calculate_overall_sensation(sensation_local_array):
+    """Calculates overall sensation.
+
+    Parameters:
+    sensation_local_array (array): Array of sensations from various body parts.
+
+    :returns
+    overall_sensation_dict (dict): Dict ....
```

Comment:
could you please better describe the output?

Reply:
Sorry about this. Added the description.